en/index.md

```diff
@@ -524,6 +524,42 @@ For example:
 
 - **Don’t use:** Click the ![pencil](https://assets.ubuntu.com/v1/633aab2b-pencil.png) pencil icon to edit the post.
 
+## Hyperlinks
+
+Here are some pointers about the general use of hyperlinks and how to format them correctly.
+
+### General use
+
+Avoid excessive links in the same paragraph or instruction. If you find yourself introducing several links in your content, consider listing them in a separate section called "Related topics", "Additional resources", or similar.
+
+When linking to versioned files or specific lines of code, copy the **permalink** instead of the URL if available. This will ensure the link is bound to the current revision of the file, so it will direct to the same content even if the file changes.
+
+### Formatting
+
+Try to make the link text match the title or heading that you are referencing. Make sure either the link text itself or the surrounding sentence provides enough context about the contents of the linked section. Avoid phrases like "this document", "this article", or "click here" as the link text.
+
+For example, when referring to a section called "Formatting":
+
+- Use: `See the [formatting guidelines](#formatting) for hyperlinks.`
+- Use: `See the [Formatting](#formatting) section for guidelines about hyperlink formatting.` 
+- Avoid: `See [Formatting](#formatting).`
+- Avoid: `See [this section](#formatting).`
+
+Avoid using a URL as the linked text.
+
+- Use: `[Page title](https://page-url.com)`
+- Don't use: `[https://page-url.com](https://page-url.com)`
+
+When possible, avoid external links. They can easily become outdated, making the document harder to maintain. Otherwise, make sure to inform the user the link is external to the current doc set by specifying the source. 
```

Comment:
I might rephrase this a little. What we want is to avoid superfluous external links. I know you know this, but sometimes they are the definitive reference for something (e.g. upstream projects).